selfdrive/car/honda/carstate.py

```diff
@@ -260,18 +259,17 @@ def update(self, cp, cp_cam, cp_body):
     else:
       ret.cruiseState.speed = cp.vl["CRUISE"]["CRUISE_SPEED_PCM"] * CV.KPH_TO_MS
 
-    self.brake_switch = cp.vl["POWERTRAIN_DATA"]["BRAKE_SWITCH"] != 0
     if self.CP.carFingerprint in HONDA_BOSCH_ALT_BRAKE_SIGNAL:
       ret.brakePressed = cp.vl["BRAKE_MODULE"]["BRAKE_PRESSED"] != 0
     else:
       # brake switch has shown some single time step noise, so only considered when
       # switch is on for at least 2 consecutive CAN samples
-      # panda safety only checks BRAKE_PRESSED signal
-      ret.brakePressed = bool(cp.vl["POWERTRAIN_DATA"]["BRAKE_PRESSED"] or
-                              (self.brake_switch and self.brake_switch_prev and cp.ts["POWERTRAIN_DATA"]["BRAKE_SWITCH"] != self.brake_switch_prev_ts))
+      # brake switch rises earlier than brake pressed but is never 1 when in park
+      if cp.updated["POWERTRAIN_DATA"]:
+        brake_switch = cp.vl["POWERTRAIN_DATA"]["BRAKE_SWITCH"] != 0
+        self.brake_switch_on = self.brake_switch_on + 1 if brake_switch else 0
```

Comment:
Occasionally we get two `POWERTRAIN_DATA` messages on one 100Hz cycle, and then the next cycle we don't get any causing update status to be False and brakePressed to fall in openpilot, while panda keeps it True until it gets an update.

Need to keep a counter, as panda only updates its previous value when updated and has no way of knowing if something isn't updated or not as it's fed messages as they come in.